server/routes/organizations/gigs/gig-controller.js

```diff
@@ -1,87 +1,171 @@
-  },
+  createGig: function (req, res){
+    var gigParams = {
+      title: req.body.title,
+      type: req.body.type,
+      date: req.body.date,
+      start_time: req.body.startTime,
+      end_time: req.body.endTime,
+      complexity: req.body.complexity,
+      health: req.body.health,
+      OrganizationId: req.organization_id,
+      LocationId: req.body.location,
+      AttireId: req.body.attire
+    };
+
+    Gig.createGig(gigParams).then(function (gig) {
+      res.send(gig);
+    }).catch(function (err) {
+      console.log(err);
+    });
 
-  getGigInfo: function(req, res){
-    res.send('getting gig info for ' + req.params.gig_id + " for org " + req.organization_id);
+   console.log("creating new gigs for organization id: " + req.params.organization_id);
   },
 
-  updateGigInfo: function(req, res){
-    res.send('updating gig info for ' + req.params.gig_id + " for org " + req.organization_id);
+  getGigInfo: function (req, res){
+    var gigId = req.params.gig_id;
+
+    Gig.getGigInfo(gigId).then(function (gig) {
+      res.send(gig);
+    }).catch(function (err) {
+      console.log(err);
+    });
+    console.log('getting gig info for gig id: ' + req.params.gig_id + " for organization id:  " + req.organization_id);
   },
 
-  deleteGig: function(req, res){
-    res.send('deleting gig ' + req.params.gig_id + " for org " + req.organization_id);
+  updateGigInfo: function (req, res){
+   var gigId = req.params.gig_id;
+   var gigParams = {
+      title: req.body.title,
+      type: req.body.type,
+      date: req.body.date,
+      start_time: req.body.startTime,
+      end_time: req.body.endTime,
+      complexity: req.body.complexity,
+      health: req.body.health,
+      OrganizationId: req.params.organization_id,
+      LocationId: req.body.location,
+      AttireId: req.body.attire
+    };
+
+    Gig.updateGigInfo(gigId, gigParams).then(function (gig) {
+      res.send(gig);
+    }).catch(function (err) {
+      console.log(err);
+    });
+    console.log('updating gig information for gig id: ' + req.params.gig_id + " for organizaiton id: " + req.organization_id);
   },
 
-  getGigStaff: function(req, res){
-    res.send('getting staff for ' + req.params.gig_id + " for org " + req.organization_id);
+  deleteGig: function (req, res){
+    var gigId = req.params.gig_id;
+
+    Gig.deleteGig(gigId).then(function (result) {
+      res.sendStatus(204);
+    }).catch(function (err) {
+      console.log(err);
+    });
+    console.log('deleting gig id:' + req.params.gig_id + " for organization id: " + req.organization_id);
   },
 
-  addEmployeeToGigStaff: function(req, res){
-    var gig_id = req.params.gig_id;
-    var employeeID = parseInt(req.body.employeeID, 10);
-    var group = req.body.group;
+  getGigStaff: function (req, res){
+    var employees = []; //holds staff for a specific gig
+    var gigId = req.params.gig_id;
+
+    UserGigs.getUserGigs(gigId, function (userGigs) {
```

Comment:
does not match the interface currently in userGigs. need to implement using promise